### DISCUSSION TITLE
Historical Stats

### DISCUSSION THREAD

**Discussion by Captain0Kangaroo** [macxq/foxess-ha#17] [Ideas]
It is possible to get a running monthly & yearly total like there is for daily generation?

It would be great to add these stats into a Dashboard.

**Comment by Daibutt** [comment]
See link;
https://github.com/macxq/foxess-ha/wiki/Understand-PV-string-power-generation-using-foxess-ha#4if-it-is-required-to-have-daily-totals-for-each-array-outside-of-the-ha-energy-dashboard

Replace daily with monthly or yearly

**Reply by Captain0Kangaroo** [reply]
Thanks. I thought it may be something like this. 

I've set my config as this: Does this seem right?

`sensor:
  - platform: foxess
    username: xxxx
    password: xxxx
    deviceID: xxxx
    
    # Setup Better Solar Tracking Sensors that resolve negative solar tracking  
  - platform: integration
    source: sensor.foxess_pv1_power
    name: pv1_sum
    unit_time: h
    round: 2
  - platform: integration
    source: sensor.foxess_pv2_power
    name: pv2_sum
    unit_time: h
    round: 2
    
# Setup Daily and Realtime solar trackers that resolve the negative solar tracking    
utility_meter:
    daily_pv1_sum:
        source: sensor.pv1_sum
        cycle: daily
 
    daily_pv2_sum:
        source: sensor.pv2_sum
        cycle: daily

    monthly_pv1_sum:
        source: sensor.pv1_sum
        cycle: monthly

    monthly_pv2_sum:
        source: sensor.pv2_sum
        cycle: monthly

    yearly_pv1_sum:
        source: sensor.pv1_sum
        cycle: yearly

    yearly_pv2_sum:
        source: sensor.pv2_sum
        cycle: yearly
# Setup Daily and real-time solar trackers that resolve the negative solar tracking          
template:
  - sensor:
      - name: "Solar Power Daily"
        unit_of_measurement: "kWh"
        state: "{{ states('sensor.daily_pv1_sum') | float + states('sensor.daily_pv2_sum') | float }}"
 
      - name: "Solar Power Now"
        unit_of_measurement: "kW"
        state: "{{ states('sensor.foxess_pv1_power') | float + states('sensor.foxess_pv2_power') | float }}"

      - name: "Solar Power Monthly"
        unit_of_measurement: "kWh"
        state: "{{ states('sensor.monthly_pv1_sum') | float + states('sensor.monthly_pv2_sum') | float }}"
 
      - name: "Solar Power Yearly"
        unit_of_measurement: "kWh"
        state: "{{ states('sensor.yearly_pv1_sum') | float + states('sensor.yearly_pv2_sum') | float }}"
`

**Comment by Daibutt** [comment]
That looks like it should work 😀

**Comment by Captain0Kangaroo** [comment]
It seems to work as the monthly and yearly just reset to zero.

If I wanted to setup a all time total what's the best way of logging this?

**Comment by Daibutt** [comment]
Sensor.pv1_sum should be always increasing so will be your all time value.
Ditto for pv2 etc..
